src/publisher/tests/test_api_v2_views.py

```diff
@@ -18,6 +18,172 @@
+            msg = "failed case %s %r, got: %s" % (
+                i + 1,
+                client_accepts,
+                resp.status_code,
+            )
+            assert 200 == resp.status_code, msg
+
+            msg = "failed case %s %r, got: %s" % (
+                i + 1,
+                client_accepts,
+                resp.content_type,
+            )
+            assert expected_accepted == resp.content_type, msg
+
+
+def test_unacceptable_types():
+    """lax responds with HTTP 406 when a content-type or a specific version 
+    of a content-type cannot be reconciled against actual content types and versions"""
+    cases = [
+        # 1, obsolete (poa v1)
+        "application/vnd.elife.article-poa+json; version=1",
+        # 2, cannot fulfill request (fixture is a poa, not a specific obsolete vor)
+        "application/vnd.elife.article-vor+json; version=1",
+        # 3, cannot fulfill request (fixture is a poa, not a specific valid vor)
+        "application/vnd.elife.article-vor+json; version=4",
+        # 4, cannot fulfill request (fixture is a poa, not a general vor)
+        "application/vnd.elife.article-vor+json",
+        # 5, cannot fulfill request (fixture is still a poa)
+        "application/vnd.elife.article-vor+json; version=1, application/vnd.elife.article-vor+json; version=2",
+        # 6, fictious content versions (for now)
+        "application/vnd.elife.article-vor+json; version=8, application/vnd.elife.article-vor+json; version=9",
+        # 7. fictious content versions (for now)
+        "application/vnd.elife.article-poa+json; version=9",
+        # 8. unrecognised content types
+        "application/foo.bar.baz; version=1",
+    ]
+
+    msid = 16695
+    ajson_fixture_v1 = join(base.FIXTURE_DIR, "ajson", "elife-16695-v1.xml.json")
+    poa_fixture = json.load(open(ajson_fixture_v1, "r"))["article"]
+    mock = Mock(article_json_v1=poa_fixture, status="poa")
+
+    with patch("publisher.logic.article_version", return_value=mock):
+        for i, client_accepts in enumerate(cases):
+            url = reverse("v2:article-version", kwargs={"msid": msid, "version": 1})
+            resp = Client().get(url, HTTP_ACCEPT=client_accepts)
+            msg = "failed case %s, got: %s" % (i + 1, resp.status_code,)
+            assert 406 == resp.status_code, msg
+
+
+def test_structured_abstract_not_downgraded_vor():
+    "a vor with a structured abstract cannot be downgraded to vor v3"
+    msid = 31549  # vor spec version 4, our first
+    fixture_path = join(
+        base.FIXTURE_DIR, "structured-abstracts", "elife-31549-v1.xml.json"
+    )
+    fixture = json.load(open(fixture_path, "r"))["article"]
+    mock = Mock(article_json_v1=fixture, status="vor")
+    vor_v3_ctype = "application/vnd.elife.article-vor+json; version=3"
+
+    with patch("publisher.logic.article_version", return_value=mock):
+        url = reverse("v2:article-version", kwargs={"msid": msid, "version": 1})
```

Comment:
I cannot exactly see how the test fixture data is version 4, does it relate to the `v2` or `1` in this line? If not, please ignore my comment, but I don't know what these values signify.

Reply:
yes, it's confusing.

this fixture: `/path/to/fixture-dir/structured-abstracts/elife-31549-v1.xml.json` is valid against VOR spec version 4 only - it contains a structured abstract and will fail validation in spec versions 1 to 3.

If a user requests in their 'accepts' header a spec version other than 4 (`3` in this case), we won't be able to satisfy it and would expect a HTTP 406 response.

This `"v2:article-version"` refers to the API version within Lax, because before the elife 2.0 infrastructure, lax had a different API with a different set of paths serving up a different set of article data (EIF, remember that?). When 2.0 came along I need to support both the old urls and new urls while everybody transitioned thus the new set of urls got the `v2` [namespace](https://docs.djangoproject.com/en/3.0/topics/http/urls/#url-namespaces).

The `article_json_v1` is an unfortunate naming choice of mine for a database field when we were adding the 2.0 support. The idea was that as new spec versions of the article json were added we would add another field to the article model to hold the article json valid for that version. This hasn't happened and is unlikely to happen. That field should be renamed.

Reply:
oh, and this `kwargs={"msid": msid, "version": 1}` is a parameter to the `/articles/$msid/version/$version` endpoint. In this context it's not being helpful ;) I'll remove it.

Reply:
fyi @NuclearRedeye 

The above is an example of how gnarly programs can get over time without proper maintenance. Imagine being new to the lax project and having to understand all that? Yet how does one prioritise and justify the time for a task like "rename db column" or "remove v1 api" against focused sprints? It goes into a stack, gets prioritised against everything else, sinks to the bottom, is politely taken off the board and gets forgotten. There is no epic for this. 

And this developer that is new to the project will come in with their task to do, become overwhelmed and (probably) advocate that it be torn down and rebuilt. In fact, this is exactly what happened with [elife-dashboard](https://github.com/elifesciences/elife-dashboard) and why we have a stillborn [elife-dashboard-2](https://github.com/elifesciences/elife-dashboard/tree/develop/dashboard_2).

And we see this pattern happening again with Kubernetes, github actions and Prometheus dashboards.

Reply:
@gnott I've removed the endpoint being patched so there are fewer mentions of 'version' floating around confusing matters.

Reply:
> EIF, remember that?

I've forgotten it sometimes, but now you mention it, I remember. The EIF support was removed on the `elife-bot` side of things, but it was kicking around for a while beyond where it was actually in use.

Your extra details here cleared up much of my confusion, and if you simplified it a little, all the better for understanding.